lib/Sema/TypeCheckStorage.cpp

```diff
@@ -2392,8 +2392,11 @@ PropertyWrapperBackingPropertyInfoRequest::evaluate(Evaluator &evaluator,
 
   // Form the initialization of the backing property from a value of the
   // original property's type.
+  Type origValueType = var->getType();
+  if (var->isInnermostPropertyWrapperInitUsesEscapingAutoClosure())
+    origValueType = FunctionType::get({}, origValueType);
```

Comment:
Another place were we could use that new API on `VarDecl` to avoid repeating this logic.